### PR TITLE
SSH serial: Close connection on read error

### PR DESCRIPTION
**Pull request by cfconrad**
If SUT was triggered to do a reboot, it could happen that the SSH
connection was closed, but we still use the SSH->sock() within select().
This cause a loop of check_ssh_serial() calls, as we never handle the pending event.
More infos in https://progress.opensuse.org/issues/60416

VR: I run a lot of tests, as it only occur sporadically.
http://cfconrad-vm.qa.suse.de/t6476 .. http://cfconrad-vm.qa.suse.de/t6492

It happen 4 times out of 16:
http://cfconrad-vm.qa.suse.de/t6477
http://cfconrad-vm.qa.suse.de/t6483
http://cfconrad-vm.qa.suse.de/t6490
http://cfconrad-vm.qa.suse.de/t6491

search for
```
[debug] ssh serial: unable to read: transport read (error code: -43) - closing connection
```
in **autoinst-log.txt**

### CODE REVIEW

File: backend/baseclass.pm [hunk 1]
```diff
@@ -1341,9 +1340,10 @@ sub stop_ssh_serial {
 
     my $ssh = $self->{serial};
     return undef unless $ssh;
-
+    bmwqemu::diag("Close SSH serial connection with " . $ssh->hostname);
```
Comment (okurz) [inline backend/baseclass.pm]: This is the "imperative" form and could confuse the reader to think that she would need to do anything. I guess you can state "Closing"? Also, as you are just concatenating the argument afterwards you can use single quotes.

Comment (cfconrad) [reply backend/baseclass.pm]: done

**Comment by okurz** [comment]
great, will merge. As always, as I assume you are still investigating further in this area so tests would be appreciated :)

**Comment by cfconrad** [comment]
Is something missing for merging it? 
I will add the test with a separate PR, so we get this fix online.

**Comment by okurz** [comment]
Oops, forgot to hit the button

**Pull request merged**